sdk/python/kfp/cli/compile_.py

```diff
@@ -133,14 +133,34 @@ def parse_parameters(parameters: Optional[str]) -> Dict:
     is_flag=True,
     default=False,
     help='Whether to disable type checking.')
+@click.option(
+    '--execution-caching-enabled-by-default',
+    type=click.Choice(['enabled', 'disabled'], case_sensitive=False),
+    default=None,
+    help='Enable task-level caching. Enabled by default, set it to disabled to disable caching.'
+)
 def compile_(
     py: str,
     output: str,
     function_name: Optional[str] = None,
     pipeline_parameters: Optional[str] = None,
     disable_type_check: bool = False,
+    execution_caching_enabled_by_default: Optional[bool] = None,
```

Comment:
`execution_caching_enabled_by_default: Optional[bool] = None` must be `execution_caching_enabled_by_default: True`
if you set None to a bool: you will have 3 states: None, True and False